### ISSUE TITLE
FR: warning when creating a new daily note/file

### ISSUE THREAD

**Issue by ksdavidc** [bartkessels/daily-note-calendar#37]
In the current implementation, there is no warning when clicking on a date (with no daily note) that a new note is being created. As a. there are 2 reasons I might click on a date (1. to find notes written on that date, 2. to create a daily note) and b. there is no indicator showing me for which dates there are daily notes, I feel it might be useful to:
- issue a warning when creating a new note
- offer a cmd-click or other alternative click that would allow distinction between the 2 use-cases, hopefully in a settings adjustable manner.

PS we discussed indicators elsewhere. This makes me think you really need 2 types of indicator!

cheers.

**Comment by ksdavidc** [comment]
I realize there are 3 reasons: a. find notes created without opening daily note, b. create notes, c. open notes explicitly.

**Comment by bartkessels** [comment]
That's an interesting one. I do think that the distinction between opening a daily note and listing the notes created on a date is something that should be implemented, as this will be easer to navigate your vault without opening a file you don't need at that time. But, I think the opening and creating of a file should be the same action, as it is kind of the same action. You want to open a file for a specific date, if it exists or not should not be on the top of your mind (at least, that's how I see it). If you do think that should be a different option, I can always make a setting for that (that way, we can all choose the behaviour 😄)

**Comment by ksdavidc** [comment]
I think you are right about the actions being the same. It is a bummer having the extra step, but at the same time, so is deleting a file one didn't want to create. How about settings like this:

```
click action:
- [ ] create/opens daily note for this day regardless
- [ ] does not create or open daily notes for this day
- [ ] does not create, only opens daily note if it exists
cmd-click action:
- [ ] create/opens daily note for this day regardless
- [ ] does not create or open daily notes for this day
- [ ] does not create, only opens daily note if it exists
cmd-shift-click: action
- [ ] create/opens daily note for this day regardless
- [ ] does not create or open daily notes for this day
- [ ] does not create, only opens daily note if it exists
```
with the actual keys perhaps being configurable to other combinations. This would allow the user to choose the mode that suits their needs. A bit of a hassle to code, but flexible for users. 

If I were going whole hog designwise, I would add some kind of signal of which mode is operant...like the plus sign that sometimes shows when option dragging to show one is copying not moving files (if you know what I mean).  3 icons representing the different actions showing which is in effect as one is about to click.

**Comment by bartkessels** [comment]
That makes sense, having some default click actions (thus the cmd or ctrl key pressed during a click) pre defined and letting people specify the exact actions in the settings. 

And as far your “hog” idea (I like this term 😂)I think it makes sense to have some sort of visualisation of what is going to happen when you click. That way you know what you’re going to do and what to expect.

**Comment by ksdavidc** [comment]
Of course the real question is how hard will it be to code....If you do get it done, I will be duly impressed!

**Comment by bartkessels** [comment]
Okay, so I've updated the plugin to use the following key bindings for navigation. When you press `shift` + date, it will only _select_ the date, thus for daily notes it will only list the notes created on that date.

In the settings menu, an extra toggle is added which allows you to only create new files using a _modifier_ key, which will be `alt` or `cmd`. This allows you to navigate existing notes without having to worry about creating new files you didn't want. And because of the `shift` + click action you can also navigate notes created on a date without having to open the corresponding daily note.

I hope this works for you as well. And the update will be there soon.

**Comment by bartkessels** [comment]
I know this isn't the solution which allows you to customize the _modifier_ key, or control what action should be done with what key press. But for now this will do the trick and in the future I might add the settings to customize it.

**Comment by bartkessels** [comment]
This has just been released in version 1.11.0 of the plugin, so when you update the plugin you should be able to navigate your notes without creating a new daily note or even opening a daily note.

And as always, thank you for your feedback. If you have any more ideas or bugs you run into, please let me know!

**Comment by ksdavidc** [comment]
Thank you so much for adding this. I really appreciate it. My first impressions, after using it for a day or two:

## Shift
I am waffling between which preference I prefer. On the one hand, I feel like the main purpose of the plugin in to open daily notes, so clicking should open the note, and seeing the notes on a given day is a bonus.

On the other hand seeing notes created on a certain day might be the main mode after all. In my case, for example, I only have daily notes on certain days (weekends), since I am using the daily notes for planning, not as a diary, but I create notes on any given day. I create a lot more notes than daily notes, so I might be more interested in that case that the default click should not open the note, just show the relevant notes. Having to remember to shift when my intent/action is doing something else (or, for example, note that there is a white bar (DN) there [^1]), so that I have to be careful to avoid an unwanted secondary action feels a bit unnatural to me.

The point being that there could be good arguments either way, but this is a good start. If someday you do get around to allowing a way to switch the behaviors, that might be cool.

I think once I use it in its current state I'll know which really works for me, but either way it is a significant improvement.

## Command

I like this option a lot. In my use case, most of the time I DON'T want to create a new note. When I do, it is with a specific intention that is enough to remind me to use the command key.


[^1]: this brings up a [different issue](https://github.com/bartkessels/daily-note-calendar/issues/55#issue-2761374559)
